src/java/htsjdk/samtools/util/Interval.java

```diff
@@ -159,12 +159,17 @@ public int compareTo(final Interval that) {
 
     /** Equals method that agrees with {@link #compareTo(Interval)}. */
     public boolean equals(final Object other) {
-        if (!(other instanceof Interval)) return false;
-        else if (this == other) return true;
-        else {
-            Interval that = (Interval)other;
-            return (this.compareTo(that) == 0);
+        if (this == other) {
+            return true;
         }
+        if (other == null) {
+            return false;
+        }
+        if (this.getClass() != other.getClass()) { 
```

Comment:
This is another one that might cause problems.  @lbergelson you've done a lot of refactoring to pull interval/region-like methods out into a single class; do you see changing `Interval.equals()` to not accept subclasses?